### PR TITLE
Slack notifier to build process

### PR DESCRIPTION
**Pull request by dansterenson**
## Status
Ready

## Related Issues
fixes: [45225](https://github.com/demisto/etc/issues/45225)

image

### CODE REVIEW

**Review by reutshal — CHANGES_REQUESTED** [review]
Looks good, please see my comments

File: build_utils/slack_notifier.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
+import argparse
+
+from circleci.api import Api as circle_api
+from slack_sdk import WebClient
+from slack_sdk.errors import SlackApiError
+from typing import List, Dict
+
+
+def get_circle_failed_steps(ci_token, build_number):
```
[the remark this answers is not in the document]
Comment (dansterenson) [reply build_utils/slack_notifier.py:9]: Done

File: build_utils/slack_notifier.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+                action_name = action.get('name', '')
+                if action_name != step_name:
+                    failed_steps_list.append(f'{step_name}: {action_name}')
+                else:
+                    failed_steps_list.append(f'{step_name}')
+
+    return failed_steps_list
+
+
+def create_slack_notifier(slack_token, build_url, failed_job_name, ci_token, build_number):
```
[the remark this answers is not in the document]
Comment (dansterenson) [reply build_utils/slack_notifier.py:30]: Done

File: .circleci/config.yml [hunk 3]
```diff
@@ -77,12 +77,19 @@ jobs:
...
           # taken to be the root directory of the workspace.
           root: /home/circleci/project
           # Must be relative path from root
           paths:
-            - build 
+            - build
+      - run:
+          name: Notify Slack
+          command: |
+            if [ "$CIRCLE_BRANCH" == "master" ]; then
```
[the remark this answers is not in the document]
Comment (dansterenson) [reply .circleci/config.yml]: Done

**Review by reutshal — APPROVED** [review]

**Review by reutshal — APPROVED** [review]

**Pull request merged**